### ISSUE TITLE
Internal Exception: java.lang.indexOutofBoundsExecption:readerIndex(13)+ length(4) exceeds writerIndex(14):UnpooledBytebufAllocator$InstrumentedUnpooledUnsafeheapBytebuf(ridx:13 widx:14 cap:14)

### ISSUE THREAD

**Issue by Semigato** [Reascer/WeaponsOfMinecraft-1.18.2#90]
when using antitheus,satsujin, and torment the game crashes

**Comment by melekaiah** [comment]
Satsujin, torment, etc. 

I am currently testing this on both a dedicated server and a local server, the local server seems good, but not the dedicated one, 

I have had this with certain skills as well, but since the error is generic, it has been hard to track down exactly what case is causing it because there "seems" to be many causes. 

sometimes its dragging axe, or dragin satsujin, 
sometimes its mid combo on the tridentstaff thingy, sometimes its while alting/dodging, sometimes its trying to active the special move. 

im careful to control parity of addons, so its always functional. 

how do i help?

**Comment by Reascer** [comment]
![image](https://github.com/Reascer/WeaponsOfMinecraft-1.18.2/assets/48482533/8a7fcaf0-900a-45fd-9174-72b66c8a70c3)
read boy ..........

**Comment by melekaiah** [comment]
Ill post everything from my end when im back from new years celeb. 

that being said. there are no crash logs, server keeps running, only client gets dc'ed, and because of that doesnt actually log anything. i cant find any logs for server for this bug. or it mentioned in the logs locally either. (shrugs) so im not sure how to post that. 

will post modlist soon. I have extensivly tested with a few of hte epic addons to rule them out. 

right now the issues are not happening on singleplayer, only on dedicated server. i have a few things left to test. 

ill post everything from my end later. 
happy new year.

**Comment by Semigato** [comment]
Yes it doesn't log anything, I'm just gonna wait until this is fixed Thanks.

**Comment by endlessMethod** [comment]
Perhaps the error is related to this.

mc: 1.20.1
forge: 47.2.18
epic fight: 20.6.3
wom: 20.1.7.36.3

full log: [debug.log](https://github.com/Reascer/WeaponsOfMinecraft-1.18.2/files/13839940/debug.log)

```
[Netty Client IO #6/DEBUG] [io.netty.channel.AbstractChannelHandlerContext/]: An exception java.lang.NullPointerException: Cannot invoke "java.lang.Throwable.printStackTrace(java.io.PrintStream)" because "cause" is null
	at MC-BOOTSTRAP/io.netty.common@4.1.82.Final/io.netty.util.internal.ThrowableUtil.stackTraceToString(ThrowableUtil.java:44)
	at TRANSFORMER/spititout@1.0.1/com.mrbysco.spititout.SpitItOut.logThrowable(SpitItOut.java:29)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.network.Connection.handler$zgk000$spititout_exceptionCaught2(Connection.java:539)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.network.Connection.exceptionCaught(Connection.java:148)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:381)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.codec@4.1.82.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.handler@4.1.82.Final/io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at MC-BOOTSTRAP/io.netty.transport@4.1.82.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at MC-BOOTSTRAP/io.netty.common@4.1.82.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at MC-BOOTSTRAP/io.netty.common@4.1.82.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:833)
was thrown by a user handler's exceptionCaught() method while handling the following exception:
java.lang.IndexOutOfBoundsException: readerIndex(13) + length(4) exceeds writerIndex(14): PooledUnsafeDirectByteBuf(ridx: 13, widx: 14, cap: 14)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1442) ~[netty-buffer-4.1.82.Final.jar%23121!/:4.1.82.Final]
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:809) ~[netty-buffer-4.1.82.Final.jar%23121!/:4.1.82.Final]
	at net.minecraft.network.FriendlyByteBuf.readInt(FriendlyByteBuf.java:1237) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at yesman.epicfight.skill.SkillDataManager$ValueType$IntegerType.readFromBuffer(SkillDataManager.java:180) ~[EpicFight-20.6.3.jar%23188!/:20.6.3]
	at yesman.epicfight.skill.SkillDataManager$ValueType$IntegerType.readFromBuffer(SkillDataManager.java:157) ~[EpicFight-20.6.3.jar%23188!/:20.6.3]
	at yesman.epicfight.network.server.SPModifySkillData.fromBytes(SPModifySkillData.java:35) ~[EpicFight-20.6.3.jar%23188!/:20.6.3]
	at net.minecraftforge.network.simple.IndexedMessageCodec.lambda$tryDecode$0(IndexedMessageCodec.java:107) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:107) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at net.minecraftforge.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:153) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at net.minecraftforge.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:63) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:260) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:252) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2387!/:?]
	at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:80) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:80) ~[forge-1.20.1-47.2.18-universal.jar%23195!/:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_7413_(ClientPacketListener.java:1863) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:57) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:7) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at net.minecraft.network.Connection.m_129517_(Connection.java:174) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:159) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:50) ~[client-1.20.1-20230612.114412-srg.jar%23190!/:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:280) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.82.Final.jar%23122!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[netty-handler-4.1.82.Final.jar%23124!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

**Comment by Zh1Ron** [comment]
Yeah i got the same error, it keeps repeating whenever i try to use a skill, hit or dodge. I tried getting highest version of forge, reconnecting, redownloading the mods and everything else. My friend is also using it yet he can run it with no problem. This is my latest log. I hope y'all can help
[latest.log](https://github.com/Reascer/WeaponsOfMinecraft-1.18.2/files/13863997/latest.log)

**Comment by Zh1Ron** [comment]
[latest.log](https://github.com/Reascer/WeaponsOfMinecraft-1.18.2/files/13864028/latest.log)
This is my log and i'm having the same issue, my friend runs it with no problem and i could as well but suddenly it gives that error which makes me unable to play

**Comment by melekaiah** [comment]
So I have some conflicting experiences here. 

On the one hand, with the exact same mod pack. running locally on my own server: no errors. 

running the exact same mod list (copy pasted) to the server folder (making sure to NOT run any client side mods on server) 
Im getting errors on a few things. 
errors include:
putting torment in my hotbar and selecting it, will kick me out instaltnly. 
the fancy two handed sword will crash when doing the special stab into ground attack. 
I also found a skill book that crashes.

these three things ONLY happen when running the modpack on a dedicated server. which makes me think the problem is there. 
I also tried to simply deleted character files from server in case they were broken, 

in all of these cases, this error pops up, 

Why is the error message so generic? what does it mean? the modlist wont help you and there is nothing in log files from dedicated server to show me anything. since its just disconnecting me. the server keeps running, so it seems to be client side, that causes a discconect to server. again, this does not happen if i just load a singleplayer world. the errors dissapear. and i can play to my hearts content.
